### DISCUSSION TITLE
Diet Management Website

### DISCUSSION THREAD

**Discussion by deepak0405** [scaleracademy/open-source-projects#58] [Ideas]
**Summary**

If you are looking for source of proteins/vitamins at google, you will find long blogs having a lot of details.  You will most likely find something similar to 

![image](https://user-images.githubusercontent.com/25735175/120923584-0a763580-c6ed-11eb-860b-474ed26f42b3.png)

The issue I find with those blogs are:
1. There is a lot of information to read
2. They have a lot of variety of food items available, out of which some items are not available to me. I want to be able to filter out the items.
3. To figure out what all food I should eat to fulfil my daily requirement, I will have to manually do the calculation.
4. If I want to optimise my diet such that my Vitamins, Proteins and Fat consumption is as per the daily requirement. Then I will have to visit multiple blogs.

To solve this problems, we can have consolidate page where you can search all the sources of proteins and can even plan your diet.

To get a rough idea of the solution, we can use this mocks

![image](https://user-images.githubusercontent.com/25735175/120923736-b586ef00-c6ed-11eb-82fe-3de7be69ff3b.png)

-----

**Features**
For the MVP, I am only planning to list all the source of nutrients. The ability to create your own diet will come after the MVP phase

1. An intuitive UI to view the sources of nutrients
    The user won't have to read through multiple long blogs to figure out the diet he is looking for. We will provide him a simple UI where he can browse through all the sources and get more details about that food.

2. Option to filter out food items
    The user can filter out the food item based on his region, his diet choices ( veg/non-veg), special diet options (keto, low carbs)
    
3. Plan your diet to get sufficient protein, vitamin and fat
    The user can select various food items and add it to his diet. Using this feature he can track and plan his daily nutrient consumption.

-----

**Future Ideas**
1.  Option to create diet plan ( not a part of MVP)
2.  A dashboard to track the diet content of the user
3.  Auto food suggestion to the user based on what his missing in his diet

-----

**User Stories**

1. Create the mockups for
     a. Each protein/vitamin page
     b. User signup page
     c. Option to add the new diet
     
2. Add the first page for proteins where we provide paginated list of all the foods providing protein

3. Add the option to search the food for protein

4. Add pages for other types of nutrients.

5. (Optional) If we want the information to be crowd sourced, so that any one can add details about the food items, then we should add the option to add/update food options.
     
-----

**Comment by sowmya-kelam** [comment]
I liked the idea more like customised  diet plan ,  including vitamin and calories count, 
we can add basic food cal and basic math for calculating  over all calories

**Reply by deepak0405** [reply]
Thanks for the suggestions!!
